File: NeetCode/Stack/C++/Stack.cpp
//
// Created by Jerry Solis on 11/16/24.
//

#include "Stack.h"

Stack::Stack(int capacity) {
    this->capacity = capacity;
    size = 0;
    arr = new int[capacity];
}

void Stack::push(int n) {
    if(size == capacity) {
        resize();
    }
    arr[size] = n;
    size++;
}

void

void peek() {
    return
}

int Stack::pop() {

}

int Stack::peek() {

}

Stack::~Stack() {
    delete[] arr;
}

File: NeetCode/Stack/Python/Stack.py
class Stack:
    def __init__(self):
        self.stack = []

    def push(self, data):
        self.stack.append(data)

    def pop(self):
        top = self.stack.pop()
        return top

    def peek(self):
        return self.stack[-1]

    def print(self):
        print(self.stack)